lib/microservice-call.js

```diff
@@ -162,74 +166,109 @@ class MicroServiceCall {
+		const { endpoint, httpMethod } = await this._getEndpointData(service, namespace, method);
+		return this._makeRequest(endpoint, httpMethod, requestData, requestHeaders, endpointParameters);
 	}
 
 	/**
-	 * Make a post request to a microservice
-	 * @deprecated
-	 * @param  {String} service The name of the microservice.
-	 * @param  {String} namespace The namespace of the microservice.
-	 * @param  {String} method The method of microservice.
-	 * @param  {Object} requestData The data that will send
-	 * @param  {Object} requestHeaders The headers of the request
-	 * @param  {Object} endpointParameters A key value to replace variables in API path
-	 * @return {Promise<MicroServiceCallResponse>}
+	 * Makes an HTTP GET request with x-janis-method 'list' to the specified service/namespace with the specified filters. Throws an Error if Services response statusCode 400+
+	 * @param {String} service The name of the microservice
+	 * @param {String} namespace The namespace of the microservice
+	 * @param {Object} filters The query params to filter/order the list
+	 * @returns {Object} Returns the response, in the body the full list of objects
 	 */
+	async list(service, namespace, filters = null) {
+
+		const body = [];
+		let page = 1;
+		let totals = 0;
+		let response;
+
+		do {
+			response = await this.call(service, namespace, 'list', filters, { 'x-janis-page': page });
 
-	post(service, namespace, method, requestData, requestHeaders, endpointParameters) {
-		return this.call(service, namespace, method, requestData, requestHeaders, endpointParameters);
+			page++;
+			totals = response.headers['x-janis-total'];
+
+			body.push(...response.body);
+		}
+		while(totals - body.length > 0);
+
+		return {
+			...response,
+			body
+		};
 	}
 
 	/**
-	 * Make a put request to a microservice
-	 * @deprecated
-	 * @param  {String} service The name of the microservice.
-	 * @param  {String} namespace The namespace of the microservice.
-	 * @param  {String} method The method of microservice.
-	 * @param  {Object} requestData The data that will send
-	 * @param  {Object} requestHeaders The headers of the request
-	 * @param  {Object} endpointParameters A key value to replace variables in API path
-	 * @return {Promise<MicroServiceCallResponse>}
+	 * Makes an HTTP GET request with x-janis-method 'list' to the specified service/namespace with the specified filters. NOT throws an Error if Services response statusCode 400+
+	 * @param {String} service The name of the microservice
+	 * @param {String} namespace The namespace of the microservice
+	 * @param {Object} filters The query params to filter/order the list
+	 * @returns {Object} Returns the response, in the body the full list of objects
 	 */
+	async safeList(service, namespace, filters = null) {
+
+		const body = [];
+		let page = 1;
+		let totals = 0;
+		let response;
+
+		do {
+			response = await this.call(service, namespace, 'list', filters, { 'x-janis-page': page });
```

Comment:
Consulta, si este metodo es safe, no deberia usar `this.safeCall`? porque `this.call` va throwear si falla 🤔